src/Traits/FireflyUser.php

```diff
@@ -17,4 +18,20 @@ public function posts(): HasMany
     {
         return $this->hasMany(Post::class);
     }
+
+    public function watching(): BelongsToMany
+    {
+        return $this->belongsToMany(Discussion::class);
+    }
+
+    public function isWatching(Discussion $discussion): bool
+    {
+        $watching = $this
+            ->watching()
+            ->select('discussion_id')
+            ->get()
+            ->pluck('discussion_id');
+
+        return $watching->contains($discussion->id);
```

Comment:
Depending on where/how often this is used we can improve this.
```suggestion
        return $this
            ->watching()
            ->where('discussion_id', $discussion->id)
            ->exists();
```

Reply:
Cleaner, thanks, I have updated the logic here.